### PR TITLE
Code review: templates (1st)

### PR DESCRIPTION
**Pull request by th4inquiry**
1. Add code examples for templates
2. Update README.md to add references

### CODE REVIEW

File: cpp1st/week02/taehwan/Templates.cpp [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+void TemplateExample::Test3() {
+    TestExample_3<double> test;
+    cout << test.TestFunc() << endl;
+}
+
+void TemplateExample::Test4() {
+    int sum_int = Add<int>(3, 4);
+    cout << sum_int << endl;
+
+    char * sum_char = Add<char *>("Hello ", "World!");
```
Comment (hmlearning) [inline cpp1st/week02/taehwan/Templates.cpp:238]: why don't you use smart pointer?

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f26-use-a-unique_ptrt-to-transfer-ownership-where-a-pointer-is-needed
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#r11-avoid-calling-new-and-delete-explicitly

File: cpp1st/week02/taehwan/Templates.cpp [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+// Copyright 2022 PentaDevs
+// Author: Taehwan Kim
+// Contents: Examples of template in C++
+
+
+#include <iostream>
+#include <memory>
+#include <cstring>
+
+#define FUNC_DEF(func) { func, #func },
```
Comment (hmlearning) [inline cpp1st/week02/taehwan/Templates.cpp:11]: test code?

**Review by hmlearning — APPROVED** [review]

**Review by BMPark-Lab — APPROVED** [review]
looks good

File: cpp1st/week02/taehwan/Templates.cpp (hunk 1 above)
Comment (th4inquiry) [reply cpp1st/week02/taehwan/Templates.cpp:236]: I will use a smart pointer next time by referring to your links, thanks!

**Pull request merged**